int64 x
int64 y
---
int64 nextX
int64 nextY
